### PR TITLE
Use DOTNET_ variables in tests

### PR DESCRIPTION
**Pull request by am11**
Renamed `COMPlus_` to `DOTNET_` under src/tests, except for configuration tests.

### CODE REVIEW

**Comment by am11** [comment]
@ViktorHofer I used neutral infrastructure label as src/tests has tests for both runtimes, and DOTNET_ / COMPlus_ variables are handled by both with separate implementations (DOTNET_ is checked first and COMPlus is a legacy fallback, hence this PR).

**Comment by AaronRobinsonMSFT** [comment]
@am11 This is great. One concern I have is in superpmi - see below. I never updated this path because I wasn't sure all the scenarios and just haven't gotten back to it. @BruceForstall Can you provide some details on when the following is used? Does the JIT team use superpmi with the test suite that is being updated here?

https://github.com/dotnet/runtime/blob/65561786b4c07d487bf9c0e7c4f16d6a6d6ad7ed/src/coreclr/tools/superpmi/superpmi/jithost.cpp#L10-L46

**Comment by am11** [comment]
@AaronRobinsonMSFT, those are defined in `superpmi.py` which I haven't changed.

File: src/tests/JIT/Performance/RunBenchmarks/RunBenchmarks.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@
 //     -listtags                  prints a list of the tag names and does nothing else
 //     -listexes                  prints a list of full path names to executables in benchmarks
 //     -runner                    run benchmarks on using runner(e.g. corerun, default is DesktopCLR)
-//     -complus_version <version> run benchmarks on particular DesktopCLR version
```
Comment (BruceForstall) [inline src/tests/JIT/Performance/RunBenchmarks/RunBenchmarks.cs:27]: I don't think you should change this file (especially the complus_version stuff). Actually, I think this whole directory is dead code that should be removed.

File: src/tests/JIT/superpmi/superpmicollect.cs [hunk 2]
```diff
@@ -696,11 +696,11 @@ private static int Main(string[] args)
 
             // Done with argument parsing.
 
-            // If someone already has one of the COMPlus_JitName/DOTNET_JitName/COMPlus_JitPath/DOTNET_JitPath variables set,
+            // If someone already has one of the DOTNET_JitName/DOTNET_JitName/DOTNET_JitPath/DOTNET_JitPath variables set,
```
Comment (BruceForstall) [inline src/tests/JIT/superpmi/superpmicollect.cs:699]: Don't change this line

File: src/tests/JIT/superpmi/superpmicollect.cs [hunk 3]
```diff
@@ -696,11 +696,11 @@ private static int Main(string[] args)
...
             // We don't want to override that. Perhaps someone is already doing a SuperPMI collection and invokes this
             // program as part of a full test path in which this program exists.
 
-            string[] checkVars = { "COMPlus_JitName", "DOTNET_JitName", "COMPlus_JitPath", "DOTNET_JitPath" };
+            string[] checkVars = { "DOTNET_JitName", "DOTNET_JitName", "DOTNET_JitPath", "DOTNET_JitPath" };
```
Comment (BruceForstall) [inline src/tests/JIT/superpmi/superpmicollect.cs:703]: Don't change this line

**Review by BruceForstall — CHANGES_REQUESTED** [review]
It's nice to move everything to our current recommended format.

I worry about how we can properly test this. We could easily lose coverage and not notice it.

In addition, there are other places that set/use COMPlus variables, e.g., src\coreclr\scripts, https://github.com/dotnet/jitutils

File: src/tests/baseservices/TieredCompilation/BasicTestWithMcj.csproj [hunk 4]
```diff
@@ -22,12 +22,12 @@
...
 mkdir r2r
 
-REM Suppress some COMPlus variables for the duration of Crossgen2 execution
+REM Suppress some DOTNET variables for the duration of Crossgen2 execution
 setlocal
-set "COMPlus_GCName="
-set "COMPlus_GCStress="
-set "COMPlus_HeapVerify="
-set "COMPlus_ReadyToRun="
+set "DOTNET_GCName="
```
[the remark this answers is not in the document]
Comment (am11) [reply src/tests/baseservices/TieredCompilation/BasicTestWithMcj.csproj:27]: Suppressed names with both prefixes.

File: src/tests/JIT/Performance/RunBenchmarks/RunBenchmarks.cs (hunk 1 above)
Comment (am11) [reply src/tests/JIT/Performance/RunBenchmarks/RunBenchmarks.cs:27]: Deleted RunBenchmarks directory.

File: src/tests/JIT/superpmi/superpmicollect.cs (hunk 2 above)
Comment (am11) [reply src/tests/JIT/superpmi/superpmicollect.cs]: Reverted.

File: src/tests/JIT/superpmi/superpmicollect.cs (hunk 3 above)
Comment (am11) [reply src/tests/JIT/superpmi/superpmicollect.cs]: Reverted.

**Comment by am11** [comment]
> In addition, there are other places that set/use COMPlus variables, e.g., src\coreclr\scripts, https://github.com/dotnet/jitutils

Those scripts can be updated separately after #77025 is merged.

**Comment by danmoseley** [comment]
Dotnet/diagnostics also has a bunch, but the docs in some cases may apply also to older runtimes.

**Review by BruceForstall — APPROVED** [review]
LGTM

**Comment by BruceForstall** [comment]
@AaronRobinsonMSFT Do you want to review?

**Comment by am11** [comment]
@BruceForstall, https://github.com/dotnet/runtime/pull/77025 is merged, I have pushed superpmi scripts change here. Also, there were few redundant exports which I've cleaned up. Please take another look. :)

**Review by BruceForstall — APPROVED** [review]
Additional changes LGTM; thanks for doing that.

**Pull request merged**